tests/NewCypress/cypress/pages/paginationPage.ts

```diff
@@ -2,52 +2,67 @@ class PaginationPage {
 
     public checkImAtTheCorrectUrl(urlPageName: string): this {
         cy.url().should('include', urlPageName);
-
         return this;
     }
 
     public returnToHome(): this {
-        const homeButton = () => cy.get('.dfe-header__link')
-
+        const homeButton = () => cy.get('.dfe-header__link'); 
         homeButton().click();
-
         return this;
     }
 
-    public next(): this {
-        cy.get(`next-page`).click();
+    public getPagination(): Cypress.Chainable<JQuery<HTMLElement>> {
+        return cy.get('.govuk-pagination'); 
+    }
 
-        return this;
+    public getPaginationButton(index: number): Cypress.Chainable<JQuery<HTMLElement>> {
+        return cy.get(`[data-testid="page-${index}"]`); 
     }
 
-    public hasNoNext(): this {
-        cy.get("next-page").should("not.exist");
+    public getTotalPaginationButtons(): Cypress.Chainable<number> {
+        return cy.get('[data-testid^="page-"]').its('length');
+    }
 
+    public clickPageNumber(pageNumber: number): this {
+        const getPageNumberButton = () => this.getPaginationButton(pageNumber);
+        getPageNumberButton().click();
         return this;
     }
 
-    public previous(): this {
-        cy.get(`previous-page`).click();
-
+    public clickNext(): this {
+        const getNextButtonElement = () => cy.contains('Next'); 
+        getNextButtonElement().click();
         return this;
     }
 
-    public hasNoPrevious(): this {
-        cy.get("previous-page").should("not.exist");
-
+    public clickPrevious(): this {
+        const getPreviousButtonElement = () => cy.contains('Previous'); 
+        getPreviousButtonElement().click();
         return this;
     }
 
-    public goToPage(pageNumber: string) {
-
-        cy.get(`page-${pageNumber}`).click();
+    public shouldNextButtonBeDisabled(): this {
```

Comment:
This (and `shouldPreviousButtonBeDisabled`) don't follow the "check" naming convention.

Reply:
Now updated this and also the next button function to follow the above convention